sp_api/auth/access_token_client.py

```diff
@@ -25,6 +25,8 @@ class AccessTokenClient(BaseClient):
     path = '/auth/o2/token'
 
     def __init__(self, refresh_token=None, credentials=None, proxies=None, verify=True):
+        if not isinstance(credentials, CredentialProvider):
```

Comment:
Please consider that `credentials` can be `None`, and it's even the argument's default value.